File: ncDados/dNotaFiscalItem.vb
Namespace nsNotaFiscalFornecedor

    Public Class ColecaoNotaFiscalItem
        Inherits List(Of dNotaFiscalItem)
    End Class

    Public Class dNotaFiscalItem
        Private _produtos_cid As Nullable(Of Integer)
        Private _produtos_descricao As String
        Private _produtos_estoque As String
        Private _produtos_valor As Decimal
        Private _produtos_referencia As String
        Private _item As Nullable(Of Integer)
        Private _caracteristicas_cid As Nullable(Of Integer)
        Private _caracteristicas_nome As String
        Private _caracteristicas_codigo As String

        Public Property produtos_cid() As Nullable(Of Integer)
            Get
                Return _produtos_cid
            End Get
            Set(ByVal value As Nullable(Of Integer))
                _produtos_cid = value
            End Set
        End Property

        Public Property produtos_descricao() As String
            Get
                Return _produtos_descricao
            End Get
            Set(ByVal value As String)
                _produtos_descricao = value
            End Set
        End Property

        Public Property produtos_estoque() As String
            Get
                Return _produtos_estoque
            End Get
            Set(ByVal value As String)
                _produtos_estoque = value
            End Set
        End Property

        Public Property item() As Nullable(Of Integer)
            Get
                Return _item
            End Get
            Set(ByVal value As Nullable(Of Integer))
                _item = value
            End Set
        End Property

        Public Property caracteristicas_cid() As Nullable(Of Integer)
            Get
                Return _caracteristicas_cid
            End Get
            Set(ByVal value As Nullable(Of Integer))
                _caracteristicas_cid = value
            End Set
        End Property

        Public Property caracteristicas_nome() As String
            Get
                Return _caracteristicas_nome
            End Get
            Set(ByVal value As String)
                _caracteristicas_nome = value
            End Set
        End Property

        Public Property caracteristicas_codigo() As String
            Get
                Return _caracteristicas_codigo
            End Get
            Set(ByVal value As String)
                _caracteristicas_codigo = value
            End Set
        End Property

        Public Property Produtos_Valor() As Decimal
            Get
                Return _produtos_valor
            End Get
            Set(ByVal value As Decimal)
                _produtos_valor = value
            End Set
        End Property

        Public Property Produtos_Referencia() As String
            Get
                Return _produtos_referencia
            End Get
            Set(ByVal value As String)
                _produtos_referencia = value
            End Set
        End Property

    End Class

End Namespace

File: Modelos/dNotaFiscalItem.cs
using System;
using System.Collections.Generic;

namespace Modelos
{
    public class ColecaoNotaFiscalItem : List<dNotaFiscalItem>
    {
    }

    public class dNotaFiscalItem
    {
        private int? _produtos_cid;
        private string _produtos_descricao;
        private string _produtos_estoque;
        private decimal _produtos_valor;
        private string _produtos_referencia;
        private int? _item;
        private int? _caracteristicas_cid;
        private string _caracteristicas_nome;
        private string _caracteristicas_codigo;

        public int? produtos_cid
        {
            get => _produtos_cid;
            set => _produtos_cid = value;
        }

        public string produtos_descricao
        {
            get => _produtos_descricao;
            set => _produtos_descricao = value;
        }

        public string produtos_estoque
        {
            get => _produtos_estoque;
            set => _produtos_estoque = value;
        }

        public int? item
        {
            get => _item;
            set => _item = value;
        }

        public int? caracteristicas_cid
        {
            get => _caracteristicas_cid;
            set => _caracteristicas_cid = value;
        }

        public string caracteristicas_nome
        {
            get => _caracteristicas_nome;
            set => _caracteristicas_nome = value;
        }

        public string caracteristicas_codigo
        {
            get => _caracteristicas_codigo;
            set => _caracteristicas_codigo = value;
        }

        public decimal Produtos_Valor
        {
            get => _produtos_valor;
            set => _produtos_valor = value;
        }

        public string Produtos_Referencia
        {
            get => _produtos_referencia;
            set => _produtos_referencia = value;
        }
    }

}
